elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/BackendRequestProcessor.java

```diff
@@ -224,6 +239,11 @@ public void shutdown() {
 			scheduler.shutdown();
 			try {
 				scheduler.awaitTermination( Long.MAX_VALUE, TimeUnit.SECONDS );
+				if ( lastAsyncWorkLatch != null ) {
+					//It's possible that a task was successfully scheduled but had no chance to run,
```

Comment:
But won't already scheduled tasks be executed prior to shutdown? That's how I take the description of `ExecutorService#shutdown()` / `awaitTermination()`.